src/console/controllers/UsersController.php

```diff
@@ -123,4 +201,28 @@ public function actionSetPassword(string $usernameOrEmail): int
 
         return ExitCode::OK;
     }
+
+    /**
+     * Creates a function for the `validator` option of `Controller::prompt`.
+     *
+     * @param Model $model
+     * @param string $attribute
+     * @param string|null $error
+     * @return callable
+     */
+    private function _createInputValidator(Model $model, string $attribute, &$error = null): callable
+    {
+        return function($input, &$error) use ($model, $attribute) {
+            $model->$attribute = $input;
+
+            if (!$model->validate([$attribute])) {
+                $error = $model->getFirstError($attribute);
+
+                return false;
+            }
+            $error = null;
+
+            return true;
+        };
+    }
```

Comment:
@brandonkelly If you're ok with this approach, we could move this somewhere reusable for [use elsewhere](https://github.com/craftcms/cms/blob/develop/src/console/controllers/InstallController.php#L272-L293).